### PR TITLE
Added simple codegen doc

### PR DESCRIPTION
**Pull request by galvesribeiro**

### CODE REVIEW

File: src/Documentation/Advanced-Concepts/Codegen.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+layout: page
+title: Code generation
+---
+
+# Code Generation
+
+**Efficient code generation is one of the pillars of Orleans Runtime**. Orleans Runtime make use of generated code in order to ensure proper serialization of types that are used across the cluster, and generating boilerplate that abstract from the Grain developer the need to understand how method shipping, exception propagation and other internal runtime concepts work.
```
Comment (ReubenBond) [inline src/Documentation/Advanced-Concepts/Codegen.md]: nit: make -> makes

Suggest reword as:

The Orleans Runtime makes use of generated code in order to ensure proper serialization of types that are used across the cluster as well as for generating boilerplate which abstracts away the implementation details of method shipping, exception propagation, and other internal runtime concepts.

File: src/Documentation/Advanced-Concepts/Codegen.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
...MSBuild` task is injected in your project's build pipeline, and the code is generated in the project's `Properties\orleans.codegen.cs`. To activate this mode, add the package `Microsoft.Orleans.OrleansCodeGenerator.Build` to your **Grain Interface** project and build it. You will notice that when adding this package no extra references are added to your project but you will see the `Properties\orleans.codegen.cs` was added. Also if you edit your `.csproj` file, you will see that an extra `MSBuild` target was added as well. The good thing of using this mode is that you can step-into the generated code while debugging since the file is physically on the disk however, your build time will be slower than usual if you have big projects. This is the default mode selected when you create a **Grain Interface** project by using the Visual Studio templates and it was the first codegen mode introduced since the release of Orleans. **Note**: To avoid the harness of keep this file under source control, we suggest you to to add it to your `.gitignore` or whatever ignore mechanism you have on your source control system, so it is not tracked by it, since this file will be re-generated every build.
```
Comment (ReubenBond) [inline src/Documentation/Advanced-Concepts/Codegen.md]: good thing of -> good thing about

physically on the disk however, -> physically on the disk. However, 

avoid the harness of keep -> avoid the hassles of keeping

File: src/Documentation/Advanced-Concepts/Codegen.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+2.	**Runtime Codegen** - introduced recently by @reubenbond, different from the build-time, this mode makes `OrleansCodeGenerator` to generate code when the Silo is starting. The good thing is that there is no `.cs` file on disk for the generated code, and you don't have an extra build step on your build pipeline. The bad thing, is that you can't step-into the generated code while debugging and it may increase your silo start time a bit. To enable this mode, if you were using the **build-time** codegen, remove all the references to `Microsoft.Orleans.OrleansCodeGenerator.Build` package and all the `Properties\orleans.codegen.cs` files then add a reference to `Microsoft.Orleans.OrleansCodeGenerator` in your **Silo host** project. If you created the project by using the Visual Studio templates, it should already be there. All set! The codegen will run at Silo startup.
```
Comment (ReubenBond) [inline src/Documentation/Advanced-Concepts/Codegen.md]: Maybe we should add a note for people who were not using build-time codegen, like:

> You can enable runtime code generation by installing the `Microsoft.Orleans.OrleansCodeGenerator` package in your silo project.

or something like that.

File: src/Documentation/Advanced-Concepts/Codegen.md (hunk 2 above)
Comment (galvesribeiro) [inline src/Documentation/Advanced-Concepts/Codegen.md]: Done

File: src/Documentation/Advanced-Concepts/Codegen.md (hunk 3 above)
Comment (ReubenBond) [inline src/Documentation/Advanced-Concepts/Codegen.md]: In the current version is says:
"If you were using the build-time codegen ..."
"If you created the project using the VS templates ..."

It doesn't yet say what happens if you just hit file -> new project and have only pulled in the core Orleans pkg into your grain interfaces project.

Comment (galvesribeiro) [inline src/Documentation/Advanced-Concepts/Codegen.md]: Done

**Pull request merged**